src/DomainLib.EventStore.Common.AspNetCore/EventStoreServiceConfigurationExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using EventStore.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+// ReSharper disable InvertIf
+
+namespace DomainLib.EventStore.Common.AspNetCore
+{
+    public static class EventStoreServiceConfigurationExtensions
+    {
+        private static readonly object SyncLock = new();
+        private static EventStoreClient _eventStoreClient;
```

Comment:
Hey Jon,

Can you help me understand why we're storing a static client and doing the lock to create it? 

Do we only want one single instance ever for the lifetime of the application?

Reply:
Yes. We only want one instance, and we can't control how many times a user might call the AddEventStore method, especially if they're using ES for events and snapshots.
If we don't do this, it would potentially overwrite the singleton in the DI container, but anyone who's already resolved it would be using a different ES client instance.
There might be nicer ways of doing this, but it does the job for now